packages/lesswrong/components/search/CommentsSearchHit.tsx

```diff
@@ -39,12 +42,24 @@ const CommentsSearchHit = ({hit, clickAction, classes, showIcon=false}: {
 }) => {
   const comment = (hit as AlgoliaComment);
   const { LWTooltip } = Components
-  const url = "/posts/" + comment.postId + "/" + comment.postSlug + "#" + comment._id
+
+  let url = "";
+  if (comment.tagSlug && comment.tagCommentType) {
+    url = tagGetCommentLink(comment.tagSlug, comment._id, comment.tagCommentType as TagCommentType);
+  } else if (comment.postId && comment.postSlug) {
+    url = `${postGetPageUrl({
+      _id: comment.postId ?? "",
+      slug: comment.postSlug ?? "",
+      isEvent: comment.postIsEvent,
+      groupId: comment.postGroupId,
+    })}#${comment._id}`;
+  }
+
   return <div className={classes.root}>
     {showIcon && <LWTooltip title="Comment">
       <ChatBubbleOutlineIcon className={classes.icon}/>
     </LWTooltip>}
-    <Link to={url} onClick={(event: React.MouseEvent) => isLeftClick(event) && clickAction && clickAction()}>
+    <Link to={url} onClick={(event: React.MouseEvent) => url && isLeftClick(event) && clickAction && clickAction()}>
```

Comment:
Why this change? I think sometimes the `clickAction` is not routing to a new page, but instead something like selecting this item, which I don't think we'd want to prevent?

Reply:
Ah hmm I wasn't aware of that. This was to prevent it routing to the wrong page if the url couldn't be constructed, but I guess that should be the responsibility of `clickAction`, I'll change this back to how it was